stumpy/mstump.py

```diff
@@ -1228,7 +1228,9 @@ def mstump(
         err = f"T is {T_A.ndim}-dimensional and must be at least 1-dimensional"
         raise ValueError(f"{err}")
 
-    core.check_window_size(m, max_size=min(T_A.shape[1], T_B.shape[1]))
+    # mstump currently supports self-join. Therefore, the argument `n` should be
+    # passed to the function `core.check_window_size`.
+    core.check_window_size(m, max_size=min(T_A.shape[1], T_B.shape[1]), n=T_A.shape[1])
```

Comment:
I'm noticing an inconsistency where sometimes we use `T.shape[1]` and sometimes we use `T.shape[-1]`. Given that `T` can only be 2 dimensional, I wonder if it is better/safer to choose `T.shape[1]`?

Reply:
I noticed it too. Do you want to address this (and any similar cases in other modules) here?

Reply:
Yes, I think it is okay to address it here in this PR.